File: apps/java-sdk/src/main/java/com/firecrawl/models/AgentTraceArtifactChange.java
package com.firecrawl.models;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.util.List;

/**
 * Describes a change to an artifact in an agent trace event.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class AgentTraceArtifactChange {

    private String kind;
    private String artifactId;
    private String path;
    private String snapshotId;
    private String change;
    private List<String> changedFields;
    private Integer itemCount;
    private String sourceToolCallId;

    public String getKind() { return kind; }
    public String getArtifactId() { return artifactId; }
    public String getPath() { return path; }
    public String getSnapshotId() { return snapshotId; }
    public String getChange() { return change; }
    public List<String> getChangedFields() { return changedFields; }
    public Integer getItemCount() { return itemCount; }
    public String getSourceToolCallId() { return sourceToolCallId; }

    @Override
    public String toString() {
        return "AgentTraceArtifactChange{kind=" + kind + ", artifactId=" + artifactId + ", change=" + change + "}";
    }
}

File: apps/php-sdk/src/Models/AgentTraceArtifactChange.php
<?php

declare(strict_types=1);

namespace Firecrawl\Models;

final class AgentTraceArtifactChange
{
    /** @param list<string>|null $changedFields */
    public function __construct(
        private readonly ?string $kind = null,
        private readonly ?string $artifactId = null,
        private readonly ?string $path = null,
        private readonly ?string $snapshotId = null,
        private readonly ?string $change = null,
        private readonly ?array $changedFields = null,
        private readonly ?int $itemCount = null,
        private readonly ?string $sourceToolCallId = null,
    ) {}

    /** @param array<string, mixed> $data */
    public static function fromArray(array $data): self
    {
        return new self(
            kind: $data['kind'] ?? null,
            artifactId: $data['artifactId'] ?? null,
            path: $data['path'] ?? null,
            snapshotId: $data['snapshotId'] ?? null,
            change: $data['change'] ?? null,
            changedFields: isset($data['changedFields']) && is_array($data['changedFields'])
                ? array_values($data['changedFields'])
                : null,
            itemCount: isset($data['itemCount']) ? (int) $data['itemCount'] : null,
            sourceToolCallId: $data['sourceToolCallId'] ?? null,
        );
    }

    public function getKind(): ?string
    {
        return $this->kind;
    }

    public function getArtifactId(): ?string
    {
        return $this->artifactId;
    }

    public function getPath(): ?string
    {
        return $this->path;
    }

    public function getSnapshotId(): ?string
    {
        return $this->snapshotId;
    }

    public function getChange(): ?string
    {
        return $this->change;
    }

    /** @return list<string>|null */
    public function getChangedFields(): ?array
    {
        return $this->changedFields;
    }

    public function getItemCount(): ?int
    {
        return $this->itemCount;
    }

    public function getSourceToolCallId(): ?string
    {
        return $this->sourceToolCallId;
    }
}
